### PR TITLE
Add a yang model for the new "Logger" table in the CONFIG DB

### PR DESCRIPTION
**Pull request by EdenGri**
Signed-off-by:  EdenGri [eden7045@gmail.com](mailto:eden7045@gmail.com)

This PR should be merged together with the following PRs as there is dependency between them:

* https://github.com/sonic-net/sonic-buildimage/pull/12544
* https://github.com/sonic-net/sonic-buildimage/pull/12553

#### Why I did it
Add the ability to the user to save the loglevel and make it persistent to reboot.

#### How I did it
Move the logger tables from the LOGLEVEL DB to the CONFIG DB. Add new yang model to verify the new config schema.

#### How to verify it
1. change the orchagent loglevel (for example) -> swssloglevel -c orchagent -l DEBUG
2. save the loglevel -> run config save
3. reboot
4. verify that the orchagent log level is still DEBUG ->run run redis-cli -n 4 hgetall "LOGGER|orchagent"

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-logger.yang [hunk 1]
```diff
@@ -0,0 +1,77 @@
+module sonic-logger{
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-logger";
```
Comment (zhangyanzhao) [inline src/sonic-yang-models/yang-models/sonic-logger.yang:5]: can we update the namespace to sonic-net ? Thanks.

File: src/sonic-yang-models/tests/yang_model_tests/tests/logger.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        "desc": "LOGGER_WRONG_LOGLEVEL pattern failure.",
+        "eStr": "wrong"
+    },
+    "LOGGER_WRONG_LOGOUTPUT": {
+        "desc": "LOGGER_WRONG_LOGOUTPUT pattern failure.",
+        "eStr": "wrong"
+    },
+    "LOGGER_MANDATORY_LOGLEVEL": {
+        "desc": "LOGGER_MANDATORY_LOGOUTPUT  no logoutput.",
+	    "eStrKey": "Mandatory"
```
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests/logger.json:15]: Add eStr: ["LOGLEVEL"]

File: src/sonic-yang-models/yang-models/sonic-logger.yang (hunk 1 above)
Comment (praveen-li) [reply src/sonic-yang-models/yang-models/sonic-logger.yang:5]: Let`s do it for all yang models later in one go.

File: src/sonic-yang-models/yang-models/sonic-logger.yang [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                leaf name {
+                    description "Component name in LOGGER table (example for component: orchagent, Syncd, SAI components).";
+                    type string;
+                }
+
+                leaf LOGLEVEL {
+                    description "The log verbosity for the component";
+                    mandatory true;
+                    type union {
+                        type swss_loglevel;
```
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-logger.yang:57]: I can still allow wrong log level to wrong component. But to resolve that we need a complex must condition, so it is up to you to add it. Thx

File: src/sonic-yang-models/yang-models/sonic-logger.yang [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    }
+                    default SYSLOG;
+                }  
+            }
+            /* end of list LOGGER_LIST */
+        }
+        /* end of LOGGER container */
+    }
+    /* end of sonic-logger container */
+}
```
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-logger.yang:76]: can we move comments in one line as:  "}  /* end of sonic-logger container */"

File: src/sonic-yang-models/tests/yang_model_tests/tests/logger.json (hunk 2 above)
Comment (EdenGri) [reply src/sonic-yang-models/tests/yang_model_tests/tests/logger.json:16]: Done, thanks

File: src/sonic-yang-models/tests/yang_model_tests/tests/logger.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+    "LOGGER_CORRECT_LOGLEVEL_AND_LOGOUTPUT": {
+        "desc": "LOGGER_CORRECT_LOGLEVEL_AND_LOGOUTPUT_1 no failure."
+    },
+    "LOGGER_WRONG_LOGLEVEL": {
+        "desc": "LOGGER_WRONG_LOGLEVEL pattern failure.",
+        "eStr": "wrong"
```
[the remark this answers is not in the document]
Comment (EdenGri) [reply src/sonic-yang-models/tests/yang_model_tests/tests/logger.json]: Done, thanks

File: src/sonic-yang-models/tests/yang_model_tests/tests/logger.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    "LOGGER_WRONG_LOGOUTPUT": {
+        "desc": "LOGGER_WRONG_LOGOUTPUT pattern failure.",
+        "eStr": "wrong"
```
[the remark this answers is not in the document]
Comment (EdenGri) [reply src/sonic-yang-models/tests/yang_model_tests/tests/logger.json]: Done, thanks

File: src/sonic-yang-models/tests/yang_model_tests/tests/logger.json (hunk 2 above)
Comment (EdenGri) [reply src/sonic-yang-models/tests/yang_model_tests/tests/logger.json:16]: Done, thanks

File: src/sonic-yang-models/yang-models/sonic-logger.yang (hunk 4 above)
Comment (EdenGri) [reply src/sonic-yang-models/yang-models/sonic-logger.yang]: Done, thanks

**Comment by EdenGri** [comment]
The tests that failed are:
bgp/test_traffic_shift.py::test_TSA[vlab-03]
tacacs/test_rw_user.py::test_rw_user[vlab-08]

File: src/sonic-yang-models/yang-models/sonic-logger.yang (hunk 4 above)
Comment (zhangyanzhao) [reply src/sonic-yang-models/yang-models/sonic-logger.yang]: @EdenGri  please click Resolve conversation button to mark this as resolved.

**Comment by qiluo-msft** [comment]
Could you refine your PR title?

File: src/sonic-yang-models/tests/yang_model_tests/tests/logger.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        "eStr": ["wrong", "LOGLEVEL"]
+    },
+    "LOGGER_WRONG_LOGOUTPUT": {
+        "desc": "LOGGER_WRONG_LOGOUTPUT pattern failure.",
+        "eStr": ["wrong", "LOGOUTPUT"]
+    },
+    "LOGGER_MANDATORY_LOGLEVEL": {
+        "desc": "LOGGER_MANDATORY_LOGOUTPUT  no logoutput.",
+        "estr": ["LOGLEVEL"],
+	    "eStrKey": "Mandatory"
```
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests/logger.json:16]: plz correct indentation.

**Review by praveen-li — APPROVED** [review]
LGTM, Thx

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by EdenGri** [comment]
The test that failed is:
container_checker/test_container_checker.py::test_container_checker[vlab-03-None-lldp]

**Comment by EdenGri** [comment]
failed in the test: configlet/test_add_rack.py::test_add_rack
in the previous running, this test was passed

**Comment by EdenGri** [comment]
failed in the test: test_bgp_update_timer[vlab-03]
in the previous running, this test was passed

**Pull request merged**